symengine/parser.cpp

```diff
@@ -545,15 +561,15 @@ class ExpressionParser
         op_stack.push(std::make_pair(-1, s_len));
         right_bracket.push(s_len);
 
-        for (int i = s_len - 1; i >= 0; i--) {
-            if (is_single_character_operator(i)
-                or is_dual_character_operator(i)) {
+        for (int i = numeric_cast<int>(s_len - 1); i >= 0; i--) {
```

Comment:
You can use `for(unsigned i = s_len;  i-->0;) {` here and get rid of casts below